### ISSUE TITLE
DXVK For Guild Wars 2 on Windows 10 / Configuration Settings question

### ISSUE THREAD

**Issue by seanmacky** [doitsujin/dxvk#1648]
Hi Doitsujin thanks for all the work you've done on DXVK !

I'm using DXVK For Guild Wars 2 on Windows 10 with Reshade and it's working great. Is there a way to configure specific settings in DXVK ? There is no config file that i can see ? Is there a way i can create a config file ?

I apologise for the noobie question !

**Comment by WinterSnowfall** [comment]
> Is there a way to configure specific settings in DXVK ? There is no config file that i can see ? Is there a way i can create a config file ?

Drop [this file](https://github.com/doitsujin/dxvk/blob/master/dxvk.conf) next to the game's executable and configure away. It's covered here: https://github.com/doitsujin/dxvk/wiki/Configuration.

**Comment by K0bin** [comment]
What kinds of graphics settings do you want to change? It should work fine out if the box.

**Comment by seanmacky** [comment]
Hi thanks for the replies.
There are two settings id like to query.

dxvk.numCompilerThreads   By default it sets mine to 7.  Do these correlate to the amount of cores I have ? Will I increase my FPS if I manually set a higher figure ? I Have a i7 6800k so 6c/ 12 threads  with HT 

 d3d9.maxAvailableMemory = 4096 

I have a RX5700XT should I set this to 8192 as I have 8 GB of VRAM 

Thank you :)

**Comment by Joshua-Ashton** [comment]
> d3d9.maxAvailableMemory = 4096
> I have a RX5700XT should I set this to 8192 as I have 8 GB of VRAM

No, the max for memory reporting in D3D9 is a UINT anyway... :<
It's okay cause it isn't tracked by default, we only do that for special games.
You can absolutely use more than 4GB VRAM in DXVK D3D9.

**Comment by K0bin** [comment]
> dxvk.numCompilerThreads =

Will only change the number of threads that are used to compile cached shaders in the background. It doesn't impact performance except on very low end CPUs. If you set it too high, those threads could potentially compete with the game threads or the DXVK cs thread for CPU time. DXVK automatically picks a number that's reasonable, so just don't worry about it.

> d3d9.maxAvailableMemory = 4096
This setting basically only exists for broken games. If you don't touch it, DXVK will report the actual amount of VRAM your GPU has.

So in conclusion: don't worry about DXVKs config options. Most of them only exist to work around broken games. If an option is necessary to run a game correctly, DXVK usually applies that automatically after someone points that out here.

**Comment by seanmacky** [comment]
Guys thank you for all of your answers and advice.

Can i expect the same level of performance when using DXVK on Windows 10 playing GW2 compared to Linux ?

**Comment by seanmacky** [comment]
Hi !  I tested performance on both the latest version of ubuntu with all performance options & tweaks enabled. GW2 does perform a lot better on Windows 10 with DXVK.   I wanted to ask are there any additional settings i can tweak in the config file ?  The reason is the FPS tanks when there's more than 10 + character models on the screen. I have a 6mb DVK cache file already established.

This behaviour does not occur using megai's d912pxy although i find his pxy adds more latency then DXVK.

Thanks for your hard work !

**Comment by doitsujin** [comment]
Please be more specific, what kind of performance are we talking about when "fps tanks"?

Should be noted that d912pxy runs pretty much nothing but GW2 and cuts a lot of corners whereas dxvk is a more or less complete D3D9 implementation, but it should still be competitive with Windows Dx9.

**Comment by seanmacky** [comment]
Yes no problem. I'm getting fps drops of 10+ frames when there's more than 10 other players on the screen. In that regards I'm getting slightly lower performance than native dx9

**Comment by doitsujin** [comment]
can you test older dxvk versions? maybe there was a regression.

**Comment by seanmacky** [comment]
Yep I'll do that. Is it worthwhile testing d9vk builds prior to the merge with dxvk as well?, 😊

**Comment by narf203** [comment]
Hi seanmacky how did you get reshade to work on win10 dxvk?
I now have gw2 dxvk working with Radial Mounts and ArcDPS but I can't get reshade to work.
Did you install vulkan or dx9 version of reshade?
And where did you put the reshade files? Inside bin64?
Did you rename any dll files for it to work?
I hope you can help me :)

**Comment by sicu** [comment]
> I now have gw2 dxvk working with Radial Mounts and ArcDPS but I can't get reshade to work.
> Did you install vulkan or dx9 version of reshade?
> And where did you put the reshade files? Inside bin64?
> Did you rename any dll files for it to work?
> I hope you can help me :)

If you want to make your life easier, use [GW2 Addon Manager](https://github.com/fmmmlee/GW2-Addon-Manager).
In this program you select ArcDPS, GW2-Radial and SelectRenderer.
Look at [SelectRenderer](https://github.com/megai2/SelectRenderer) for where to drop the various render files and reshade files.
After that start up the game and a popup window will let you select the combo you want to run.
Hope that helps :)

**Comment by narf203** [comment]
Thanks I got it working now.
SelectRender wasn't showing the menu at first. After putting dxgi.dll in bin64 and starting the game I got the menu.
Closed the game removed dxgi.dll in bin64 and now It's all working perfectly.
